### ISSUE TITLE
Empty space for Only Me categories

### ISSUE THREAD

**Issue by tuxpiper** [ushahidi/platform#4743]
How to reproduce:
Prerequisite steps
1. Login to your deployment
2. Click on "Settings"
3. Click on "Categories"
4. Create a category, set the visibility to "Only me"
5. Go back to "Settings"
6. Select "Surveys"
7. Create a survey or edit an old one
8. Click on "Add new field"
9. Select "Categories"
10. Select the category you just created
11. Save the survey
12. Add a new post to the survey
13. Make sure the category you added is selected
14. Click submit post
15. Go to feed-view
16. Publish the post

Reproduce the issue
As a logged out user,
1. Go to feed view
2. Locate the post you just created and click on it

**Expected behaviour**
Categories that the current user cannot see is not visible in any way under the category-field

**Actual behaviour**
There is a tag-icon with an empty space after it for the category that you just added.

Let's remove the marker (green circle) and not show any indication that a category is there but hidden.
<img width="979" alt="Screenshot 2023-10-02 at 16 01 59" src="https://github.com/ushahidi/platform/assets/8624777/3cd93cfc-4d8e-465b-a621-40ee3c5a5198">
<img width="980" alt="Screenshot 2023-10-02 at 16 03 20" src="https://github.com/ushahidi/platform/assets/8624777/c8bd8cd5-9a19-4e92-9201-6c44771e6158">

---
Created automatically from USH-899 (https://linear.app/ushahidi/issue/USH-899/empty-space-for-only-me-categories)

**Comment by Chiemezuo** [comment]
Hi!! This is the only free issue I've seen for now. I'll be giving it a go.

**Comment by Chiemezuo** [comment]
@Angamanga

**Comment by Angamanga** [comment]
Thank you @Chiemezuo! I will review shortly!
